extensions/wikidata/src/org/openrefine/wikidata/editing/EditBatchProcessor.java

```diff
@@ -149,33 +154,47 @@ public void performEdit()
-                descriptions.addAll(update.getDescriptionsIfNew().stream()
-                        .filter(desc -> !currentDocument.getDescriptions().containsKey(desc.getLanguageCode())).collect(Collectors.toList()));
-                editor.updateTermsStatements(currentDocument,
-                		labels,
-                        descriptions,
-                        update.getAliases().stream().collect(Collectors.toList()),
-                        new ArrayList<MonolingualTextValue>(),
-                        update.getAddedStatements().stream().collect(Collectors.toList()),
-                        update.getDeletedStatements().stream().collect(Collectors.toList()),
-                        summary, tags);
+                EntityIdValue newCell = (EntityIdValue) update.getItemId();
+                if (newCell instanceof ItemIdValue) {
+	                ItemDocument currentDocument = (ItemDocument) currentDocs.get(update.getItemId().getId());
+	                List<MonolingualTextValue> labels = update.getLabels().stream().collect(Collectors.toList());
+	                labels.addAll(update.getLabelsIfNew().stream()
+	                      .filter(label -> !currentDocument.getLabels().containsKey(label.getLanguageCode())).collect(Collectors.toList()));
+	                List<MonolingualTextValue> descriptions = update.getDescriptions().stream().collect(Collectors.toList());
+	                descriptions.addAll(update.getDescriptionsIfNew().stream()
+	                        .filter(desc -> !currentDocument.getDescriptions().containsKey(desc.getLanguageCode())).collect(Collectors.toList()));
+	                editor.updateTermsStatements(currentDocument,
+                            labels,
+	                        descriptions,
+	                        update.getAliases().stream().collect(Collectors.toList()),
+	                        new ArrayList<MonolingualTextValue>(),
+	                        update.getAddedStatements().stream().collect(Collectors.toList()),
+	                        update.getDeletedStatements().stream().collect(Collectors.toList()),
+	                        summary, tags);
+                } else if (newCell instanceof MediaInfoIdValue) {
+                    MediaInfoDocument currentDocument = (MediaInfoDocument) currentDocs.get(update.getItemId().getId());
+	                List<MonolingualTextValue> labels = update.getLabels().stream().collect(Collectors.toList());
+	                labels.addAll(update.getLabelsIfNew().stream()
+	                      .filter(label -> !currentDocument.getLabels().containsKey(label.getLanguageCode())).collect(Collectors.toList()));
+	                MediaInfoUpdate updatesCollection = Datamodel.makeMediaInfoUpdate((MediaInfoIdValue) update.getItemId(), currentDocument.getRevisionId(), Datamodel.makeTermUpdate(labels, tags), Datamodel.makeStatementUpdate(update.getAddedStatements(), update.getDeletedStatements(), tags));
```

Comment:
This line is too long: you should introduce intermediate variables for the arguments. For instance, by writing on the line above:
`TermUpdate labelUpdate = Datamodel…`

Also, it seems that you have put too much trust in your IDE's auto-completion mechanism. When you use the IDE to guess the arguments of a method, it will try to find objects in the context with matching types. Sometimes that works well, in other cases that is nonsense. For instance, the variable `tags` has type `List<String>` and it looks like your IDE has used it for all the places where a `List<String>` is expected. But those `tags` are just tags that are applied when the edit is made (they are not part of the content that we are pushing), so it does not make sense to use `tags` in any part of the `MediaInfoUpdate`.

So I would encourage you to have a critical eye on the suggestions made by the IDE. When using a method, you can look at the documentation of that method to understand what the required arguments are supposed to mean, and check if the suggested variables make sense for that.